### PR TITLE
[AzureMonitorExporter] onboard to aot ci

### PR DESCRIPTION
**Pull request by TimothyMothra**
This PR is to onboard to the AOT CI. https://github.com/Azure/azure-sdk-for-net/blob/main/doc/dev/AotRegressionChecks.md

A regression in AOT compatibility was reported in #44127
- 1.3.0-beta.1 supports AOT
- 1.3.0-beta.2 does not

The root cause was a change in the Azure.Core dependency.
To help catch regressions sooner, we need to onboard our Exporter to the AOT CI introduced in #40629

### CODE REVIEW

File: doc/dev/AotRegressionChecks.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ extends:
     CheckAOTCompat: true
     AOTTestInputs:
     - ArtifactName: [Name of package]
-      ExpectedAOTWarningsFilePath: None [or filepath of errors relative to the service directory]
+      ExpectedWarningsFilePath: None [or filepath of errors relative to the service directory]
```
Comment (m-redding) [inline doc/dev/AotRegressionChecks.md:35]: Thank you! 😅

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ILC : Trim analysis warning IL2026: [Azure.Core]Azure.Core.Serialization.DynamicData: Using member 'Azure.Core.Serialization.DynamicData.DynamicDataJsonConverter.DynamicDataJsonConverter()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Using DynamicData or DynamicDataConverter is not compatible with trimming due to reflection-based serialization.
```
Comment (m-redding) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt:1]: If you get the latest from main you should be able to just delete this file pending: https://github.com/Azure/azure-sdk-for-net/pull/44208 
I don't _think_ there should be any warnings reported with .NET 8

File: sdk/monitor/ci.yml [hunk 3]
```diff
@@ -47,3 +47,7 @@ extends:
       safeName: AzureMonitorQuery
     - name: Azure.Monitor.Ingestion
       safeName: AzureMonitorIngestion
+    CheckAOTCompat: true
+    AOTTestInputs:
+    - ArtifactName: Azure.Monitor.OpenTelemetry.Exporter
+      ExpectedWarningsFilepath: /Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt
```
Comment (m-redding) [inline sdk/monitor/ci.yml:53]:
```suggestion
      ExpectedWarningsFilepath: None
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt:1]: Thanks for reviewing. I'll wait till 44208 merges and then refresh this PR. 
Please let me know if you need something from me more urgently!

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/aotcompatibility/ExpectedAotWarnings.txt:1]: I got latest from main branch. The AOT check is running and the error count is down to 4!

FYI: MacOS ci is failing:
> error NETSDK1204: Ahead-of-time compilation is not supported on the current platform 'osx-x64'.

**Comment by TimothyMothra** [comment]
Good news, Build is showing 0 warnings!
image


Bad news, MacOS builds are failing

> ##[error]/Users/runner/hostedtoolcache/dotnet/sdk/8.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): Error NETSDK1204: Ahead-of-time compilation is not supported on the current platform 'osx-x64'.

**Comment by m-redding** [comment]
It's coming from this I believe - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj#L4 I'm not exactly sure how to fix it, I tried to when I was testing the pipeline in https://github.com/Azure/azure-sdk-for-net/pull/44302/files but I couldn't get the targets right

**Comment by TimothyMothra** [comment]
> It's coming from this I believe - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj#L4 I'm not exactly sure how to fix it, I tried to when I was testing the pipeline in https://github.com/Azure/azure-sdk-for-net/pull/44302/files but I couldn't get the targets right

Oh I missed that it was _my_ AOT project. Weird that it only fails on this PR. Regardless, I think I can fix this. Thank you!

**Comment by TimothyMothra** [comment]
@m-redding thanks for all your help! Can you please review and sign off on these changes?

**Review by m-redding — APPROVED** [review]
This all looks good! So glad to see the warnings are resolved. And thanks for fixing the md

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**